core/src/main/scala/org/http4s/Message.scala

```diff
@@ -267,6 +267,28 @@ final class Request[F[_]](
 
   def queryString: String = uri.query.renderString
 
+  /** cURL representation of the request.
+    *
+    * Supported Parameters are:
+    *
+    * -X,
+    * -H (redacts sensitive information such as Authorization- or Cookie-Headers)
+    */
+  def asCurl: String = {
```

Comment:
I have worked in a parameter for this.
You meant something like that?
Or should it get more into detail whether to log headers after all and when to redact them?

Reply:
The `Logger` middleware takes a [`CaseInsensitiveString => Boolean`](https://github.com/http4s/http4s/blob/28945bbb4ba0c4c9122c2bac3af2e3333f2bfb64/server/src/main/scala/org/http4s/server/middleware/Logger.scala#L25).  It would be great to stay consistent with that, since it's the same basic problem.